### PR TITLE
Will

### PR DESCRIPTION
**Pull request by WillHHH**

### CODE REVIEW

File: week5/will/QQzone/index.css [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+.login-container {
+  width: 100%;
+  max-width: 880px;
+  position: absolute;
+  top: 95px;
+}
+
+.login-container img {
+  width: 250px;
+  height: 99px;
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.css:28]: why is the height 99 px?

File: week5/will/QQzone/index.css [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  top: 0px;
+  right: 0;
+  font-family: "microsoft Yahei", "微软雅黑", SimHei, "黑体", "Hiragino Sans GB";
+}
+
+.login-block h3 {
+  color: #333;
+  font-size: 20px;
+  font-weight: 200;
+  margin: 5px 0px 15px 0px;
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.css:51]: you don't need the last 0 in this case, use: margin: 5px 0px 15px;

File: week5/will/QQzone/index.css [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+.login-input-container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+}
+
+.login-input-container input {
+  width: 258px;
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.css:72]: can you not use a fixed width input? can this be 100%?

File: week5/will/QQzone/index.css [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  height: 18px;
+  font-size: 13px;
+  font-family: Verdana, Tahoma, Arial;
+  color: #eee;
+}
+
+.login-input-container button {
+  background-color: #86ce30;
+  color: #fff;
+  width: 272px;
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.css:87]: do we need to have a fixed width button?

File: week5/will/QQzone/index.css [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+.login-footer a {
+  font-size: 12px;
+  color: #666;
+}
+
+.login-footer span {
+  font-weight: 200;
+  font-size: 10px;
+  margin: 0px 3px;
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.css:130]: can we use an even number for margin?

File: week5/will/QQzone/index.css [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+  flex-direction: column;
+}
+
+.footer-logo-container span:hover {
+  transition: all 0.1s linear;
+  transform: translateY(-8px)
+}
+
+.footer-logo-container .winphone p {
+  margin-bottom: -13px;
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.css:171]: why are we using a negative margin here?

File: week5/will/QQzone/index.html [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      <div id="login-block" class="login-block">
+        <h3>帐号密码登录</h3>
+        <p>推荐使用<a onclick="handleClick()" href="#">快速安全登录</a>，防止盗号。</p>
+        <div class="login-input-container">
+          <input type="text" placeholder="支持QQ号/邮箱/手机号登录" />
+          <input type="password" placeholder="密码" />
+          <button type="submit">登 录</button>
+        </div>
+        <div class="login-footer">
+          <a href="">忘了密码？</a>
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.html:27]: use a hashtag instaed of an empty href

File: week5/will/QQzone/index.html [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        <h3>快速安全登录</h3>
+        <p>请使用<a href="#">QQ手机版</a>扫描二维码，</p>
+        <p>安全登录，防止盗号。</p>
+        <img src="QEcode.png" alt="qecode">
+        <div class="login-footer">
+          <a onclick="handleSwitchBack()" href="">帐号密码登录</a>
+          <span>|</span>
+          <a href="">注册新账号</a>
+          <span>|</span>
+          <a href="">意见反馈</a>
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.html:44]: same as above

File: week5/will/QQzone/index.html [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+          <a href=""><img src="winphone.png" alt="logo" /></a>
+          <p>Windows Phone</p>
+        </span>
+        <span>
+          <a href=""><img src="infocheck.png" alt="logo" /></a>
+          <p>其他手机</p>
+        </span>
+      </div>
+      <div class="footer-text">
+        <div class="footer-text-firstline">
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.html:72]: do not use names such as firstline

File: week5/will/QQzone/index.html [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+          <a href="#">社交组件</a>
+          <span>|</span>
+          <a href="#">应用侵权投诉</a>
+          <span>|</span>
+          <a href="#">Complaint Guidelines</a>
+          <span>|</span>
+        </div>
+        <p>
+          Copyright © 2005 - 2020 Tencent.
+          <a href="" target="">All Rights Reserved</a>
```
Comment (mxdi9i7) [inline week5/will/QQzone/index.html:98]: empty href => href="#"

File: week5/will/taobao/index.css [hunk 11]
```diff
@@ -0,0 +1,296 @@
...
+  height: 395px;
+  margin-top: 35px;
+  z-index: 0;
+  position: relative;
+}
+
+.login-container img {
+  width: 200px;
+  position: absolute;
+  right: 3px;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css:64]: even number

File: week5/will/taobao/index.css [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+  top: 15px;
+  z-index: 1;
+}
+
+.login-block {
+  z-index: 2;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: there's no position for this element, why are you using zindex?

File: week5/will/taobao/index.css [hunk 13]
```diff
@@ -0,0 +1,296 @@
...
+}
+
+.login-form-input {
+  margin-bottom: 20px;
+  position: relative;
+}
+
+.login-form-input i {
+  width: 40px;
+  height: 41px;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: why are you using 41 as a number for px?

File: week5/will/taobao/index.css [hunk 14]
```diff
@@ -0,0 +1,296 @@
...
+  background-color: #ddd;
+  color: #fff;
+  font-size: 20px;
+  position: absolute;
+  top: 1px;
+}
+
+.login-form-input input {
+  width: 248px;
+  height: 37px;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: even number

File: week5/will/taobao/index.css [hunk 15]
```diff
@@ -0,0 +1,296 @@
...
+  padding-left: 48px;
+  font-size: 14px;
+  font-weight: 200;
+}
+
+.login-block form button {
+  width: 301px;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: even number

File: week5/will/taobao/index.css [hunk 16]
```diff
@@ -0,0 +1,296 @@
...
+  height: 43px;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: same

File: week5/will/taobao/index.css [hunk 17]
```diff
@@ -0,0 +1,296 @@
...
+  position: relative;
+  border-radius: 3px;
+  padding-left: 20px;
+  margin-top: 20px;
+}
+
+.login-button i {
+  font-size: 18px;
+  position: absolute;
+  left: 3px;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: even

File: week5/will/taobao/index.css [hunk 18]
```diff
@@ -0,0 +1,296 @@
...
+footer {
+  width: 990px;
+  margin: auto;
+  padding: 15px 0px;
+}
+
+.footer-firstline a {
+  padding: 0px 10px 0px 5px;
+  border-right: 1px solid #485065;
+  margin-left: -2px;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: why use a negative value?

File: week5/will/taobao/index.css [hunk 19]
```diff
@@ -0,0 +1,296 @@
...
+}
+
+.footer-thirdline p {
+  color: #9c9c9c;
+  font-family: "Microsoft YaHei", tahoma, arial, "Hiragino Sans GB",
+    "\5b8b\4f53", sans-serif;
+  font-size: 12px;
+  display: inline;
+  padding: 0px 10px 0px 6px;
+  margin-left: -4px;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: negative

**Review by mxdi9i7 — CHANGES_REQUESTED** [review]
Missing the hover effect for the phone to appear

File: week5/will/QQzone/index.css (hunk 1 above)
Comment (WillHHH) [reply week5/will/QQzone/index.css:28]: fixed

File: week5/will/QQzone/index.css (hunk 2 above)
Comment (WillHHH) [reply week5/will/QQzone/index.css:51]: fixed

File: week5/will/QQzone/index.css (hunk 3 above)
Comment (WillHHH) [reply week5/will/QQzone/index.css:72]: fixed

File: week5/will/QQzone/index.css (hunk 4 above)
Comment (WillHHH) [reply week5/will/QQzone/index.css:87]: fixed

File: week5/will/QQzone/index.css (hunk 5 above)
Comment (WillHHH) [reply week5/will/QQzone/index.css:130]: fixed

File: week5/will/QQzone/index.css (hunk 6 above)
Comment (WillHHH) [reply week5/will/QQzone/index.css:171]: fixed

File: week5/will/QQzone/index.html (hunk 7 above)
Comment (WillHHH) [reply week5/will/QQzone/index.html:27]: fixed

File: week5/will/QQzone/index.html (hunk 8 above)
Comment (WillHHH) [reply week5/will/QQzone/index.html:44]: fixed

File: week5/will/QQzone/index.html (hunk 9 above)
Comment (WillHHH) [reply week5/will/QQzone/index.html:72]: fixed

File: week5/will/QQzone/index.html (hunk 10 above)
Comment (WillHHH) [reply week5/will/QQzone/index.html:98]: fixed

File: week5/will/taobao/index.css (hunk 11 above)
Comment (WillHHH) [reply week5/will/taobao/index.css:64]: fixed

File: week5/will/taobao/index.css (hunk 12 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/taobao/index.css (hunk 13 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/taobao/index.css (hunk 14 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/taobao/index.css (hunk 15 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/taobao/index.css (hunk 16 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/taobao/index.css (hunk 17 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/taobao/index.css (hunk 18 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/taobao/index.css (hunk 19 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/8/index.js [hunk 20]
```diff
@@ -0,0 +1,21 @@
+const will = {
+    name: "Will Wu",
+    birthplace: "China",
+    age: "25",
```
Comment (mxdi9i7) [inline week5/will/8/index.js]: age needs to be a number

File: week5/will/8/index.js [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+    isGraduated: "false",
+    education: "The future sphere",
```
Comment (mxdi9i7) [inline week5/will/8/index.js:6]: capitalize

File: week5/will/8/index.js [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+    },
+};
+
+window.alert(will.name);
+
+document.write(will.age);
+
+document.getElementById("food").innerHTML = will.favoriteFood;
+
+console.log(will.favoriteStar);
```
Comment (mxdi9i7) [inline week5/will/8/index.js]: you are console logging an object, you have to select which property to console log

File: week5/will/9/index.js [hunk 23]
```diff
@@ -0,0 +1,108 @@
+const isString = input => {
+  if (typeof input == "string") {
+    return true;
+  } else {
```
Comment (mxdi9i7) [inline week5/will/9/index.js]: dont need the else here

File: week5/will/9/index.js [hunk 24]
```diff
@@ -0,0 +1,108 @@
...
+    return false;
+  }
+};
+
+// console.log(isString("8"));
+
+const isBlank = input => {
+  if (input.length === 0) {
+    return true;
+  } else {
```
Comment (mxdi9i7) [inline week5/will/9/index.js]: same as above

File: week5/will/9/index.js [hunk 25]
```diff
@@ -0,0 +1,108 @@
...
+
+// console.log(truncateString("Robin Hood", 4));
+
+const stringParameterize = input => input.replace(/ /g, "-").toLowerCase();
+
+// console.log(stringParameterize("Robin Hood from China"));
+
+const camelize = input => {
+  const x = input.split(" ").join("");
+  const y = x.substr(0, 1).toLowerCase();
```
Comment (mxdi9i7) [inline week5/will/9/index.js]: use a more descriptive variable name

File: week5/will/9/index.js [hunk 26]
```diff
@@ -0,0 +1,108 @@
...
+  const z = x.slice(1);
```
Comment (mxdi9i7) [inline week5/will/9/index.js]: no y and z

File: week5/will/9/index.js [hunk 27]
```diff
@@ -0,0 +1,108 @@
...
+  return y + z;
+};
+
+// console.log(camelize("JavaScript Exercises"));
+
+const randomize = (min, max) => {
+  let x = Math.random();
+  let y = min + x * (max - min);
+  return Math.floor(y);
```
Comment (mxdi9i7) [inline week5/will/9/index.js]: no x y, use real names

File: week5/will/9/index.js [hunk 28]
```diff
@@ -0,0 +1,108 @@
...
+};
+
+// console.log(randomize(10, 15));
+
+const spaceTrailing = input => {
+  const l = input.length;
```
Comment (mxdi9i7) [inline week5/will/9/index.js]: use names

File: week5/will/9/index.js [hunk 29]
```diff
@@ -0,0 +1,108 @@
...
+};
+
+console.log(phoneNumberFormating("6466397087"));
+console.log(phoneNumberFormating("16466397087"));
+console.log(phoneNumberFormating("(646)6397087"));
+console.log(phoneNumberFormating("646-639-7087"));
+console.log(phoneNumberFormating("646639-7087"));
+console.log(phoneNumberFormating("347265918"));
+console.log(phoneNumberFormating("34726591899"));
+console.log(phoneNumberFormating("(646)-639-7087"));
```
Comment (mxdi9i7) [inline week5/will/9/index.js:149]: try using for loop and write a second version of formatPhone()

File: week5/will/taobao/index.css [hunk 30]
```diff
@@ -260,30 +259,34 @@ footer {
   padding: 15px 0px;
 }
 
-.footer-firstline a {
+.footer-uperline a {
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: spelling is not correct, separate 2 words with hyphens

File: week5/will/taobao/index.css [hunk 31]
```diff
@@ -260,30 +259,34 @@ footer {
...
   padding: 0px 10px 0px 5px;
   border-right: 1px solid #485065;
-  margin-left: -2px;
+  margin-left: 0px;
 }
 
-.footer-secondline a {
+.footer-middleline a {
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css]: separate 2 words with hyphens

File: week5/will/taobao/index.css [hunk 32]
```diff
@@ -260,30 +259,34 @@ footer {
...
-.footer-thirdline p {
+.footer-lowerline p {
   color: #9c9c9c;
   font-family: "Microsoft YaHei", tahoma, arial, "Hiragino Sans GB",
     "\5b8b\4f53", sans-serif;
   font-size: 12px;
   display: inline;
   padding: 0px 10px 0px 6px;
-  margin-left: -4px;
+  margin-left: 0px;
```
Comment (mxdi9i7) [inline week5/will/taobao/index.css:290]: does it still look the same with a 0 margin to the left?

File: week5/will/taobao/index.html [hunk 33]
```diff
@@ -88,7 +88,7 @@ <h4>24小时在线帮助</h4>
         </div>
     </div>
     <footer>
-        <div class="footer-firstline infonav">
+        <div class="footer-uperline infonav">
```
Comment (mxdi9i7) [inline week5/will/taobao/index.html]: spelling

File: week5/will/taobao/index.html [hunk 34]
```diff
@@ -103,9 +103,9 @@ <h4>24小时在线帮助</h4>
             <a href="">淘宝旅行</a>
             <a href="">虾米</a>
             <a href="">来往</a>
-            <a class="border-clear" href="">支付宝</a>
+            <a class="border-clear" href="#">支付宝</a>
         </div>
-        <div class="footer-secondline infonav">
+        <div class="footer-middleline infonav">
```
Comment (mxdi9i7) [inline week5/will/taobao/index.html]: use hyphen to separate 2 words

File: week5/will/taobao/index.html [hunk 35]
```diff
@@ -119,7 +119,7 @@ <h4>24小时在线帮助</h4>
             <a href="">法律声明</a>
             <span><em>© 2003-2017 Taobaobei.com 版权所有</em></span>
         </div>
-        <div class="footer-thirdline infonav">
+        <div class="footer-lowerline infonav">
```
Comment (mxdi9i7) [inline week5/will/taobao/index.html]: hyphen

**Review by mxdi9i7 — CHANGES_REQUESTED** [review]

File: week5/will/8/index.js (hunk 20 above)
Comment (WillHHH) [reply week5/will/8/index.js]: fixed

File: week5/will/8/index.js (hunk 21 above)
Comment (WillHHH) [reply week5/will/8/index.js:6]: fixed

File: week5/will/8/index.js (hunk 22 above)
Comment (WillHHH) [reply week5/will/8/index.js]: fixed

File: week5/will/9/index.js (hunk 23 above)
Comment (WillHHH) [reply week5/will/9/index.js]: fixed

File: week5/will/9/index.js (hunk 24 above)
Comment (WillHHH) [reply week5/will/9/index.js]: fixed

File: week5/will/9/index.js (hunk 25 above)
Comment (WillHHH) [reply week5/will/9/index.js]: fixed

File: week5/will/9/index.js (hunk 26 above)
Comment (WillHHH) [reply week5/will/9/index.js]: fixed

File: week5/will/9/index.js (hunk 27 above)
Comment (WillHHH) [reply week5/will/9/index.js]: fixed

File: week5/will/9/index.js (hunk 28 above)
Comment (WillHHH) [reply week5/will/9/index.js]: fixed

File: week5/will/9/index.js (hunk 29 above)
Comment (WillHHH) [reply week5/will/9/index.js:149]: fixed

File: week5/will/taobao/index.css (hunk 30 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/taobao/index.css (hunk 31 above)
Comment (WillHHH) [reply week5/will/taobao/index.css]: fixed

File: week5/will/taobao/index.html (hunk 35 above)
Comment (WillHHH) [reply week5/will/taobao/index.html]: fixed

File: week5/will/taobao/index.css (hunk 32 above)
Comment (WillHHH) [reply week5/will/taobao/index.css:290]: fixed

File: week5/will/taobao/index.html (hunk 33 above)
Comment (WillHHH) [reply week5/will/taobao/index.html]: fixed

File: week5/will/taobao/index.html (hunk 34 above)
Comment (WillHHH) [reply week5/will/taobao/index.html]: fixed

**Review by mxdi9i7 — APPROVED** [review]

**Pull request merged**